components/popover/popover-mixin.js

```diff
@@ -62,34 +69,105 @@ export const PopoverMixin = superclass => class extends superclass {
 
 	constructor() {
 		super();
+		this.noAutoClose = false;
 		this.opened = false;
-		this._useNativePopover = isSupported;
+		this._useNativePopover = isSupported ? 'manual' : undefined;
+		this._handleAutoCloseClick = this._handleAutoCloseClick.bind(this);
+		this._handleAutoCloseFocus = this._handleAutoCloseFocus.bind(this);
 	}
 
 	connectedCallback() {
 		super.connectedCallback();
+		if (this.opened) this._addAutoCloseHandlers();
 	}
 
 	disconnectedCallback() {
 		super.disconnectedCallback();
+		this._removeAutoCloseHandlers();
 	}
 
 	updated(changedProperties) {
 		super.updated(changedProperties);
 		if (changedProperties.has('opened')) {
 
+			if (this._useNativePopover) {
+				if (this.opened) this.showPopover();
+				else this.hidePopover();
+			}
+
+			this._previousFocusableAncestor = this.opened ? getPreviousFocusableAncestor(this, false, false) : null;
+
 			if (this.opened) {
+				this._opener = getComposedActiveElement();
+				this._addAutoCloseHandlers();
 				this.dispatchEvent(new CustomEvent('d2l-popover-open', { bubbles: true, composed: true }));
 			} else if (changedProperties.get('opened') !== undefined) {
+				this._removeAutoCloseHandlers();
 				this.dispatchEvent(new CustomEvent('d2l-popover-close', { bubbles: true, composed: true }));
 			}
 
-			if (this._useNativePopover) {
-				if (this.opened) this.showPopover();
-				else this.hidePopover();
-			}
+		}
+	}
+
+	_addAutoCloseHandlers() {
+		this.addEventListener('blur', this._handleAutoCloseFocus, { capture: true });
+		document.body.addEventListener('focus', this._handleAutoCloseFocus, { capture: true });
+		document.addEventListener('click', this._handleAutoCloseClick, { capture: true });
+	}
+
+	_close() {
+		const hide = () => {
+			this.opened = false;
+		};
+
+		hide();
+	}
+
+	_handleAutoCloseClick(e) {
+
+		if (!this.opened || this.noAutoClose) return;
 
+		const rootTarget = e.composedPath()[0];
+		if (isComposedAncestor(this.shadowRoot.querySelector('.content'), rootTarget)
+			|| isComposedAncestor(this._opener, rootTarget)) {
+			return;
 		}
+
+		this._close();
+	}
+
+	_handleAutoCloseFocus() {
+
+		// timeout needed to work around lack of support for relatedTarget
```

Comment:
Is this still an issue?

Reply:
Ah, good question. I was mostly trying not to break anything when we update dropdown and tooltip to use this. 😬  I was sorta of hoping to look at switching that later after the dust settles from the rest of the changes. But I can explore using `relatedTarget` earlier rather than later if preferred.

Reply:
That's totally fair. I'm ok with this going in as-is. Maybe just call this out as a to do? Or whatever you have going for keeping track of things to come back to.

Reply:
Cool... added a todo.